app/controllers/admin/users_controller.rb

```diff
@@ -100,7 +101,29 @@ def remove_role
   private
 
   def user_params
-    params.require(:user).permit(:name, :organization_id, :email, :password, :password_confirmation)
+    organization_id = params[:user][:organization_id]
+
+    raise "Please select an organization for the user." if organization_id.blank?
+
+    user_params = params.require(:user).permit(:name, :organization_id, :email, :password, :password_confirmation)
+    user_params[:organization_role_join_attributes] = { role_id: updated_role_id(organization_id) }
+
+    user_params
+  rescue => e
+    redirect_back(fallback_location: edit_admin_user_path, error: e)
+  end
+
+  def updated_role_id(organization_id)
+    user_role_title = Role::TITLES[Role::ORG_USER]
+    user_role_type = Role::ORG_USER
+
+    role = Role.find_by(resource_type: user_role_title, resource_id: organization_id, name: user_role_type)
+
+    if role
+      role.id
+    else
+      raise "Error finding a role within the provided organization"
+    end
```

Comment:
I'll accept it, but FYI for the future, lines 122-126 can be replaced by

`role&.id || raise "Error finding a role within the provided organization"`

Reply:
Yeah okay thanks, I will update it accordingly.